doc/developers/maintainer.rst

```diff
@@ -141,36 +150,49 @@ Making a release
    The wheel building workflow is managed by GitHub Actions and the results be browsed at:
    https://github.com/scikit-learn/scikit-learn/actions?query=workflow%3A%22Wheel+builder%22
 
-   Once all works, you can proceed with tagging. Create the tag and push it (if
+.. note::
+
+  Before building the wheels, make sure that the ``pyproject.toml`` file is
+  up to date and using the oldest version of ``numpy`` for each Python version
+  to avoid ABI incompatibility issues. Moreover, a new line have to be included
```

Comment:
```suggestion
  to avoid API incompatibility issues. Moreover, a new line have to be included
```

Reply:
This is ABI for Application Binary Interface. I will make that more explicit.